### PR TITLE
Yet more js-sdk test deflakification

### PR DESCRIPTION
**Pull request by richvdh**
Requires a version of matrix-mock-request including https://github.com/matrix-org/matrix-mock-request/pull/5, so this won't build yet.

### CODE REVIEW

File: spec/integ/matrix-client-room-timeline.spec.js [hunk 1]
```diff
@@ -252,11 +252,14 @@ describe("MatrixClient room timelines", function() {
                 client.scrollback(room).done(function() {
                     expect(room.timeline.length).toEqual(1);
                     expect(room.oldState.paginationToken).toBe(null);
-                    done();
+
+                    // still have a sync to flush
+                    httpBackend.flush("/sync", 1).then(() => {
```
Comment (lukebarnard1) [inline spec/integ/matrix-client-room-timeline.spec.js]: I'm not sure why this has to flush immediately before calling `done`

Comment (richvdh) [inline spec/integ/matrix-client-room-timeline.spec.js]: See the commit comment (https://github.com/matrix-org/matrix-js-sdk/pull/499/commits/adca75b7d89d91d5d8a019e0f873a023a69d2ddb): the `afterEach` checks it's been flushed.

Comment (lukebarnard1) [inline spec/integ/matrix-client-room-timeline.spec.js]: I'm not sure the `afterEach` should be used for that, should it not just kill the httpBackend altogether?

Comment (lukebarnard1) [inline spec/integ/matrix-client-room-timeline.spec.js]: We decided (via https://riot.im/develop/#/room/#riot-dev:matrix.org/$1500047765866CodIk:sw1v.org) to leave that discussion for another day.

**Pull request merged**